From 8ab862cbd4eeeb52ab921a555b7a006e89fd2abc Mon Sep 17 00:00:00 2001
From: "Banavathu, Srinivas Naik" <snb@codeaurora.org>
Date: Fri, 21 Sep 2012 09:58:12 +0530
Subject: [PATCH 092/283] Telephony: Consider EVDO ecio zero value as valid.

When mEvdoEcio receives 0 getEvdoEcio returns -1.

Ecio valid values are positive integers, to fix this
consider EVDO ecio zero value as valid.

Change-Id: Ice8415150e7502c7f39e1a4135d35ea2bec1eb1f
CRs-Fixed: 402159
---
 telephony/java/android/telephony/SignalStrength.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/telephony/java/android/telephony/SignalStrength.java b/telephony/java/android/telephony/SignalStrength.java
index 92b889b..4671310 100755
--- a/telephony/java/android/telephony/SignalStrength.java
+++ b/telephony/java/android/telephony/SignalStrength.java
@@ -279,7 +279,7 @@ public class SignalStrength implements Parcelable {
         mCdmaEcio = (mCdmaEcio > 0) ? -mCdmaEcio : -160;
 
         mEvdoDbm = (mEvdoDbm > 0) ? -mEvdoDbm : -120;
-        mEvdoEcio = (mEvdoEcio > 0) ? -mEvdoEcio : -1;
+        mEvdoEcio = (mEvdoEcio >= 0) ? -mEvdoEcio : -1;
         mEvdoSnr = ((mEvdoSnr > 0) && (mEvdoSnr <= 8)) ? mEvdoSnr : -1;
 
         // TS 36.214 Physical Layer Section 5.1.3, TS 36.331 RRC
-- 
1.8.1.1

